src/library/EndpointManagement/EndpointManagementClient.cs

```diff
@@ -454,6 +398,22 @@ private async Task EnsureEndpointManagerRunningAsync(CancellationToken cancellat
                                                 cancellationToken: cancellationToken,
                                                 out string outPut);
 
+                    // pkexec allows an authorized user to execute PROGRAM as another user. Refer - https://linux.die.net/man/1/pkexec
+                    // if pkexec failed then launchExitCode will not be zero, so giving user another chance to try with sudo. 
+                    // But this might fail for users who don't have sudo access. This is specifically for Linux.
+                    if (launchExitCode != 0 && _platform.IsLinux && !_environmentVariables.IsCodespaces) {
+                         _log.Info($"pkexec failed with exitCode {launchExitCode}, retrying with sudo");
+                        (fileName, command) = await GetEndpointManagerLaunchArguments(currentUserName, logFileDirectory, cancellationToken);

```

Comment:
Is there a need to call `GetEndpointManagerLaunchArguments` again? Wouldn't the logic be identical if this line was removed?

Reply:
good point , updated them.